### PR TITLE
MSSQL: Add Windows AD/Kerberos auth

### LINKED ISSUE #19795: MSSQL Server. Authenticating with domain users
**What would you like to be added**:
Being able to authenticate with AD users

**Why is this needed**:
In our Corporation all MSSQL users are AD users. When we try to connect we get the following error:
Login error: mssql: Login failed. The login is from an untrusted domain and cannot be used with Windows authentication.

It's the same error you get when using jDTS if not properly configured. With that jdbc driver you have to pass de domain, user, and usentlmv2 parameters with the conection url.

### PR DESCRIPTION
**Pull request by asimpson**
**What is this feature?**

This allows users to authenticate to a MSSQL instance via Windows Active Directory (Kerberos).

**Why do we need this feature?**

Windows Active Directory authentication is an oft-requested mode of authentication for MSSQL.

**Which issue(s) does this PR fix?**:

Fixes #19795

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.

Manually testing this requires operating from a host that is domain joined. I've put together a [doc](https://docs.google.com/document/d/1uGrEYSA37Cmgc4e8k_6wFc09ZViExP6-mMu40gxlDFM/edit?usp=sharing) that explains how to test this.

MSSQL makes a distinction between users who can access the server itself and users who can access specific databases. If you are attempting to test a specific database and user, you need to make sure that user is granted access to the database. On the testing instance I've already created a test db that works: `gftest`.

The connection string is the most important bit of all this. There are tests for each connection mode.

- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by aangelisc — APPROVED** [review]
Amazing work on this! Everything LGTM and I've tested all 4 auth methods (thank you for making that extremely easy). 🎉 🏆 

The only question I have is if this would work in Grafana Cloud? I assume not so maybe it's worth adding a note to that effect?

**Comment by asimpson** [comment]
@aangelisc it won't for two reasons:
1. 3/4 methods require file paths at the moment.
2. username/password auth doesn't rely on a file path but does rely on the KDC being available publicly which is not a great security posture to have. 

I can note this in the docs, great call out.

File: docs/sources/getting-started/get-started-grafana-ms-sql-server.md [hunk 1]
```diff
@@ -28,18 +28,61 @@ If you are on a Windows host but want to use Grafana and MS SQL data source on a
...
 | `Host`     | The IP address/hostname and optional port of your MS SQL instance. If port is omitted, the default 1433 will be used. |
 | `Database` | Name of your MS SQL database.                                                                                         |
-| `User`     | Database user's login/username.                                                                                       |
-| `Password` | Database user's password.                                                                                             |
+
+##### SQL Server Authentication
+
+| Name       | Description                     |
+| ---------- | ------------------------------- |
+|            |
```
Comment (bossinc) [inline docs/sources/getting-started/get-started-grafana-ms-sql-server.md:51]:
```suggestion
```

**Pull request merged**